### DISCUSSION TITLE
Where is this storing at?

### DISCUSSION THREAD

**Discussion by Brandon-Perry** [react-native-async-storage/async-storage#668] [Q&A]
Quick question,

So far it seems to work fine, I'm able to save and get information. But one thing that's confusing me is where exactly my information is being stored. I have React Native Debugger running and I looked at the local storage tab and I don't see my information in there. Is it hidden somewhere else?

**Comment by tido64** [comment]
As far as I know, the local storage tab is just something that comes with Chrome debugger for debugging web apps. It's not meant for React Native apps. You might want to look into using [Flipper](https://fbflipper.com/) and a [plugin for AsyncStorage](https://github.com/Fausto95/rn-async-storage-flipper).

**Reply by Foxeye-Rinx** [reply]
I just got a bug in my code while working on web with expo, the `AsyncStorage.getItem()` returns mismatch with the Chrome's devtool
<img width="456" alt="image" src="https://user-images.githubusercontent.com/22351393/215597008-f06e25d4-dcf5-4ddc-b731-38753427dd1a.png">
<img width="875" alt="image" src="https://user-images.githubusercontent.com/22351393/215597146-3296494a-e1b8-4b55-b3aa-72c629e9052e.png">

The problem is when the devtool is crashed somehow by extensions or other reasons. It does not throw any error and I didn't know it was crashed already. So even tried to put a random string on the key (on chrome devtool), but actually the string was not saved into the window.localStorage and cause misleading.

**To solve:** close the chrome devtool and open again.

**Comment by tamish-max** [comment]
What you're seeing is expected: React Native AsyncStorage is **not** the same thing as the browser's `localStorage`, so it won't show up in the Chrome / React Native Debugger local-storage tab.

On React Native, the package stores data in the app's **native persistent storage** (inside your app sandbox). On web, the implementation is different again.

From your code's point of view, the important part is just that it is persistent key/value storage:

```ts
await AsyncStorage.setItem('@user', JSON.stringify(user))
const raw = await AsyncStorage.getItem('@user')
const parsed = raw ? JSON.parse(raw) : null
```

If you want to inspect what is there, a more reliable approach is:

```ts
const keys = await AsyncStorage.getAllKeys()
console.log(keys)
```

and then read specific keys with `getItem`.

For tooling, Flipper has historically been the better option for native storage inspection than the browser debugger tab.

So short version: your data is stored natively, not in the debugger's web `localStorage` view.
